test/jdk/java/awt/Window/Grab/GrabTest.java

```diff
@@ -173,28 +188,39 @@ public static void test() {
         // 5. Check that press on the other frame's title causes ungrab
         f1.setVisible(true);
         Util.waitForIdle(robot);
+
         Util.clickOnTitle(f1, robot);
+        Util.waitForIdle(robot);
+        robot.delay(100);
+
         if (!ungrabbed) {
             passed = false;
             System.err.println("Failure: [5] Press inside of other Frame's title didn't cause ungrab");
         }
         f.requestFocus(); // restore focus
         Util.waitForIdle(robot);
+        robot.delay(100);
+
         if (!f.hasFocus()) {
             System.err.println("Error: Frame can't be focused");
         }
         ungrabbed = false;
         tk.grab(w);
+        robot.delay(100);
 
 
         // 6. Check that press on the outside area causes ungrab
         Point loc = f.getLocationOnScreen();
         robot.mouseMove(loc.x + 100, loc.y + f.getSize().height + 10);
         Util.waitForIdle(robot);
+        robot.delay(100);
+
         robot.mousePress(InputEvent.BUTTON1_MASK);
-        robot.delay(50);
+        robot.delay(100);
         robot.mouseRelease(InputEvent.BUTTON1_MASK);
         Util.waitForIdle(robot);
```

Comment:
Can be replaced with BUTTON1_DOWN_MASK.

Has the fix been tested on macOS 14 ?

Reply:
That (about macOS 14) is the important question. Why is it so unstable on macOS 14 but fine everywhere else ?

Also can't you use "setAutoDelay(100)" instead of adding all these one-off delays ?